### PR TITLE
Automate psi ci for mac and windows 

### LINKED ISSUE #4408: Verify automation process to run macOS tests on PSI
/kind user-story

## User Story
As a QE I want to analyse/research and automate macOS test on PSI.

## Acceptance Criteria 
- [ ] Mention the steps needed for automation
- [ ] Automate the process of test

## Links 
- Feature Request: N/A
- Related issue: #2540 

/kind user-story

### LINKED ISSUE #4412: Automate integration and unit test for windows on PSI
/kind user-story

## User Story
As a QE I want to automate both integration and unit test on Windows.

## Acceptance Criteria 
- [ ] Automate the test for Windows.
- [ ] Resolve the blockers, if any.

## Links 
- Feature Request: N/A
- Related issue: https://github.com/openshift/odo/issues/4379 , #2540

/kind user-story

### PR DESCRIPTION
**Pull request by anandrkskd**
**What type of PR is this?**
/kind feature

**What does does this PR do / why we need it**:
This PR adds scripts required for automating windows and mac tests for PR

**Which issue(s) this PR fixes**:

Fixes - part of #4412
Fixes - part of #4408 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
N/A

### CODE REVIEW

**Comment by anandrkskd** [comment]
[ci/prow/v4.6-integration-e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_odo/4460/pull-ci-openshift-odo-master-v4.6-integration-e2e/1364567879332139008) failure is already tracked under https://github.com/openshift/odo/issues/4301
ci/prow/v4.6-e2e-4x-psi was supposed to fail
retesting

/test v4.6-integration-e2e

File: scripts/openshiftci-e2e-4x-psi-tests_mac.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-e2e-4x-psi-tests_mac.sh:6]: As per our meeting today, can we merge this and openshift-e2e-4x-psi-tests_win.sh together and pass win or mac as parameter values to set SENDQUEUE and SENDTOPIC based on that?
This will reduce the number of scripts

File: scripts/openshiftci-e2e-4x-psi-tests_win.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-e2e-4x-psi-tests_win.sh:6]: see comment for previous script

File: scripts/openshiftci-unit-psi-tests_mac.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-unit-psi-tests_mac.sh:8]: same comments as for previous script

File: scripts/openshiftci-unit-psi-tests_win.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-unit-psi-tests_win.sh:8]: Same as above

**Review by rnapoles-rh — CHANGES_REQUESTED** [review]
see comments

**Review by rnapoles-rh — COMMENTED** [review]
/lgtm

File: scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh:18]: should't be scripts/openshift**ci**-e2e-4x-psi-tests.sh ?

File: scripts/openshiftci-unit-psi-tests_win_mac.sh
Comment (rnapoles-rh) [inline scripts/openshiftci-unit-psi-tests_win_mac.sh:23]: should't be scripts/openshift**ci**-e2e-4x-psi-tests.sh ?

**Review by rnapoles-rh — COMMENTED** [review]
se comments

File: scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh
Comment (anandrkskd) [reply scripts/openshiftci-e2e-4x-psi-tests_win_mac.sh:18]: I must have missed it, I will update it ASAP

**Review by rnapoles-rh — APPROVED** [review]
/lgtm

**Comment by girishramnani** [comment]
anand suggested that I dont merge this as some tests are not active yet

**Review by rnapoles-rh — APPROVED** [review]
/LGTM

**Comment by prietyc123** [comment]
@anandrkskd I think we should not say it as fix until we have a successful job running on the prs.

**Comment by anandrkskd** [comment]
I am running the Jobs on jenkin's and will update when it passes, and fix in case of failure.

File: scripts/run_script_e2e.sh
Comment (prietyc123) [inline scripts/run_script_e2e.sh:16]: I think `run_all_tests` script https://github.com/openshift/odo/blob/master/scripts/run_all_tests.sh is setting up similar stuffs. Shouldn't we consolidate the scripts?

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/setup_script_e2e.sh
Comment (prietyc123) [inline scripts/setup_script_e2e.sh:1]: https://github.com/openshift/odo/pull/4460#discussion_r598451047

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/run_script_e2e.sh
Comment (anandrkskd) [reply scripts/run_script_e2e.sh:16]: As it was discussed with @mohammedzee1000 that run_all_tests script will be divided into two different parts, i.e. setup scripts and run scripts, we will remove `run_all_tests` when the transition is complete. For now we are not removing run_all_tests because we are still using this script for Linux tests.

File: scripts/setup_script_e2e.sh
Comment (anandrkskd) [reply scripts/setup_script_e2e.sh:1]: https://github.com/openshift/odo/pull/4460#discussion_r598454613

File: scripts/setup_script_e2e.sh
Comment (mik-dass) [inline scripts/setup_script_e2e.sh:22]: These are linux paths which are used as env values https://github.com/openshift/odo/blob/ba7161940c735caffaa9bd49e731dd15b8e9a61f/pkg/testingutil/configs.go#L19 Thus those won't work in case of windows.

Comment (anandrkskd) [reply scripts/setup_script_e2e.sh:22]: Added, Thanks @mik-dass

File: scripts/setup_script_unit.sh
Comment (prietyc123) [inline scripts/setup_script_unit.sh:20]: I think we should not overwrite the default `ARTIFACT_DIR` on prow as per our last conversation with platform team. I think you mean `ARTIFACTS_DIR`, no? Same as here https://github.com/openshift/odo/blob/master/scripts/run_all_tests.sh#L20

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/setup_script_e2e.sh
Comment (prietyc123) [inline scripts/setup_script_e2e.sh:22]: `CUSTOM_HOMEDIR=$ARTIFACTS_DIR` should be `CUSTOM_HOMEDIR=$ARTIFACT_DIR` because for custom dir has the path set by prow team. I don't think exporting https://github.com/openshift/odo/pull/4460/files#diff-13486c16e4a0760249f14a4ebfc7f573868866f417260c72093497f868584487R21 makes sense in this case.

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: scripts/setup_script_e2e.sh
Comment (prietyc123) [reply scripts/setup_script_e2e.sh:22]: I am not sure where are we using this `ARTIFACTS_DIR`

Comment (anandrkskd) [reply scripts/setup_script_e2e.sh:22]: It's kind of confusing because, ARTIFACT_DIR is set by the prow team, but we are using our PSI setup that may not work as intended on PRs.
I will remove the ARTIFACTS_DIR from the script for now and check if there is any error because of this env when triggered on PRs by prow after it gets merged . ARTIFACT_DIR is not preset if we are not using prow to trigger the test, so it will fail.
Does that sound fine for us?

Comment (prietyc123) [reply scripts/setup_script_e2e.sh:22]: >  ARTIFACT_DIR is not preset if we are not using prow to trigger the test, so it will fail.

Yes.

Comment (prietyc123) [reply scripts/setup_script_e2e.sh:22]: >  ARTIFACT_DIR is not preset if we are not using prow to trigger the test, so it will fail.

Yes.

Comment (prietyc123) [reply scripts/setup_script_e2e.sh:22]: I think you can verify your current changes on jenkin, If I am not wrong.

Comment (prietyc123) [reply scripts/setup_script_e2e.sh:22]: I think you can verify your current changes on jenkin, If I am not wrong.

Comment (anandrkskd) [reply scripts/setup_script_e2e.sh:22]: I guess we will need to add that env from jenkins side to test

**Comment by rnapoles-rh** [comment]
kubernetes integration tests failing with 2h0m0s timeout. We changed to 4hs, where is this where is this  2h0m0s timeout coming from?

**Comment by anandrkskd** [comment]
>kubernetes integration tests failing with 2h0m0s timeout. We changed to 4hrs, where is this 2h0m0s timeout coming from?

I can see the timeout i set to 4hr00min in the test, seems like this timeout was from prow. @prietyc123 can you confirm?

**Comment by prietyc123** [comment]
> I can see the timeout i set to 4hr00min in the test, seems like this timeout was from prow. @prietyc123 can you confirm?

Yes, this timeout is the default one from prow. As it fails on pre configuration of test, it will reflect the prow one. 

```
INFO[2021-05-03T10:39:44Z] Ran for 2h3m11s                              
ERRO[2021-05-03T10:39:44Z] Some steps failed:                           
ERRO[2021-05-03T10:39:44Z]   * could not run steps: step kubernetes-integration-e2e failed: "kubernetes-integration-e2e" test steps failed: "kubernetes-integration-e2e" pod "kubernetes-integration-e2e-kubernetes-integration-e2e-steps" failed: the pod ci-op-3cwblct6/kubernetes-integration-e2e-kubernetes-integration-e2e-steps failed after 2h0m27s (failed containers: test): ContainerFailed one or more containers exited 
ERRO[2021-05-03T10:39:44Z] Container test exited with code 1, reason Error 
ERRO[2021-05-03T10:39:44Z] ---                                          
ERRO[2021-05-03T10:39:44Z] Getting ci-firewall, see https://github.com,/mohammedzee1000/ci-firewall 
ERRO[2021-05-03T10:39:44Z] + curl -kLO https://github.com/mohammedzee1000/ci-firewall/releases/download/v0.1.1/ci-firewall-linux-amd64.tar.gz 
ERRO[2021-05-03T10:39:44Z]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current 
ERRO[2021-05-03T10:39:44Z]                                  Dload  Upload   Total   Spent    Left  Speed 
ERRO[2021-05-03T10:39:44Z] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   638  100   638    0     0   3067      0 --:--:-- --:--:-- --:--:--  3067 
ERRO[2021-05-03T10:39:44Z] 
100 6389k  100 6389k    0     0  10.6M      0 --:--:-- --:--:-- --:--:-- 10.6M 
```

**Review by mik-dass — APPROVED** [review]
/approve

**Comment by anandrkskd** [comment]
ci/prow/v4.7-integration-e2e  test failed due to platform issue.

/test v4.7-integration-e2e

**Pull request merged**
